Объекты на массив менять не стала.

sofka opened the pull request:
Добавила файл работы с бд firebase.
Состояние перевернутости(active) карточки покемона сохраняется в бд, после обновления страницы, карточки перевернуты.
Добавила метод добавления нового покемона, создание вынесла в отдельную функцию.
Стили для кнопки добавила.

Теперь вопросы:
Методы обращения к бд нужно выносить в отдельные файл или методы? И в нужных ли местах я их написала?
Правильно ли у меня сделан useEffect? можно ли так..Он работает, вроде правильно.

sofka committed:
Объекты на массив менять не стала.
Добавила файл работы с бд firebase.
Состояние перевернутости(active) карточки покемона сохраняется в бд, после обновления страницы, карточки перевернуты.
Добавила метод добавления нового покемона, создание вынесла в отдельную функцию.
Стили для кнопки добавила.

Теперь вопросы:
Методы обращения к бд нужно выносить в отдельные файл или методы? И в нужных ли местах я их написала?
Правильно ли у меня сделан useEffect? можно ли так..Он работает, вроде правильно.

ZarMarathon commented on src/rootes/GamePage/index.js:
```diff
@@ -1,148 +1,17 @@
+        database.ref('pokemons').once('value', (snapshot) => {
+            setPokemons(snapshot.val());
+        });
+    }, [currentPokemons]);
```
ну нет, смотри, ты внутри эффекта меняешь currentPokemons и зависишь от этой же переменной, по идее она должна у тебя вызываться бесконечно) тебе нужно, чтобы useEffect сработал один раз при первом рендере покемона, поправь))

ZarMarathon commented on src/rootes/GamePage/index.js:
```diff
@@ -152,23 +21,87 @@ const GamePage = () => {
+                const pokemon = { ...item[1] };
+                if (pokemon.id === id) {
+                    pokemon.active = !pokemon.active;
+                    database.ref('pokemons/' + objID).set(pokemon);
```
можно и так, но что, если запрос к БД не отработает?)
.set() возвращает промис, лучше обновление стейта делать в .then, при успешной записи в БД)

ZarMarathon reviewed (approved):
аппрув, но цикличный useEffect поправь) лучше не добавлять в зависимости эффекта то, что ты в этом же эффекте и меняешь))

sofka committed:
Убрала из useEffect currentPokemons.
Переделала функции поворота карточки и добавления нового покемона. Теперь сначала идет запись в бд, потом обновляется state.
